### DISCUSSION TITLE
How to sync the files in storage and the corresponding public URL in a database table?

### DISCUSSION THREAD

**Discussion by RatulSaha** [supabase/supabase#4596] [Questions]
In a project that will have images added programatically (in tens of thousands), we are considering to store the images in one bucket (structurally, multiple folders). We need to show them to the FE quickly, so plan to put the public url of the image in a text column in the corresponding database row.

Question: is there a way to get notified of changes of the actual file in storage? The worry is that if the file needs to be changed/deleted, an automated way to update the corresponding database column would be super useful.

**Comment by GaryAustin1** [comment]
You could put a trigger on the Supabase storage object table and then have a function query the url or file object id in your table and take appropriate action.
